metabase/metabase
# Fix tooltip shows data for different X value

Closes https://github.com/metabase/metabase/issues/42243

### Description

Fixes tooltip shows data for different x-value.
The dataset normalization for normalized stacked charts lost the original datum index.

### How to verify

Verify the reported question:
- Download CSV results from https://metaboat.slack.com/archives/C01LQQ2UW03/p1714764194303799
- Upload to your instance
- When you hover the last bar it shows the correct tooltip

#### A simpler example

Create a stacked bar chart from the query

```sql
select DATE '2020-01-01' x, 10 y1, 20 y2
union all select null, null, null
union all select DATE '2020-02-01', 10, 20
```

When hovering the second bar ensure the tooltip shows correct values

### Demo

Before
<img width="985" alt="Screenshot 2024-05-06 at 7 05 03 PM" src="https://github.com/metabase/metabase/assets/14301985/9c4080e9-a36c-4e2e-9b48-1f51ac2c2b8b">

After
<img width="1059" alt="Screenshot 2024-05-06 at 7 07 17 PM" src="https://github.com/metabase/metabase/assets/14301985/f31d4781-0966-45b9-9869-ebe590ad0b58">

### Checklist
